crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -166,138 +150,61 @@ impl<'a> IRGenerator<'a> {
-            return self.get_current_value(&value);
-        }
-
-        let ident_name = self.ident_name(&ident);
-        let obj = env.get(&ident_name);
-        let o_type = self.context.context.def_interner.id_type(ident.id);
-
-        let v_id = match obj {
-            Object::Array(a) => {
-                let obj_type = o_type.into();
-                //We should create an array from 'a' witnesses
-                self.context.create_array_from_object(&a, ident.id, obj_type, &ident_name)
-            }
-            _ => {
-                let obj_type = ObjectType::get_type_from_object(&obj);
-                //new variable - should be in a let statement? The let statement should set the type
-                self.context.add_variable(
-                    node::Variable {
-                        id: NodeId::dummy(),
-                        name: ident_name.clone(),
-                        obj_type,
-                        root: None,
-                        def: Some(ident.id),
-                        witness: node::get_witness_from_object(&obj),
-                        parent_block: self.context.current_block,
-                    },
-                    None,
-                )
-            }
-        };
-
-        self.context.get_current_block_mut().update_variable(v_id, v_id);
-
-        Value::Single(v_id)
-    }
-
-    pub fn def_interner(&self) -> &NodeInterner {
-        &self.context.context.def_interner
+    fn codegen_identifier(&mut self, ident: &Ident) -> Value {
+        let value = self.variable_values[&ident.id].clone();
+        self.get_current_value(&value)
     }
 
     fn codegen_prefix_expression(
         &mut self,
         rhs: NodeId,
-        op: HirUnaryOp,
+        op: UnaryOp,
     ) -> Result<NodeId, RuntimeError> {
         let rtype = self.context.get_object_type(rhs);
         match op {
-            HirUnaryOp::Minus => {
+            UnaryOp::Minus => {
                 let lhs = self.context.zero_with_type(rtype);
                 let operator = BinaryOp::Sub { max_rhs_value: BigUint::zero() };
                 let op = Operation::Binary(node::Binary { operator, lhs, rhs });
                 self.context.new_instruction(op, rtype)
             }
-            HirUnaryOp::Not => self.context.new_instruction(Operation::Not(rhs), rtype),
+            UnaryOp::Not => self.context.new_instruction(Operation::Not(rhs), rtype),
         }
     }
 
     fn codegen_infix_expression(
         &mut self,
         lhs: NodeId,
         rhs: NodeId,
-        op: HirBinaryOp,
+        op: BinaryOpKind,
     ) -> Result<NodeId, RuntimeError> {
         let ltype = self.context.get_object_type(lhs);
-        //n.b. we do not verify rhs type as it should have been handled by the type checker.
-
-        if let (HirBinaryOpKind::Assign, Some(lhs_ins)) =
-            (op.kind, self.context.try_get_mut_instruction(lhs))
-        {
-            if let Operation::Load { array_id, index } = lhs_ins.operation {
```

Comment:
Nowhere - the `HirBinaryOpKind::Assign` was impossible since assignments were not a HirBinaryOp and were their own separate node type in the Hir. The unused HirBinaryOpKind::Assign has been removed